cross builds failed last night

Might be tranisent but:

https://github.com/compiler-explorer/compiler-workflows/runs/4945268234?check_suite_focus=true

```
  [EXTRA]    Patching gcc-git-e0b8716f
  Error:   
  Error:   >>
  Error:   >>  Build failed in step 'Extracting and patching toolchain components'
  Error:   >>        called in step '(top-level)'
  Error:   >>
  Error:   >>  Error happened in: CT_DoExecLog[scripts/functions@376]
  Error:   >>        called from: CT_DoExtractPatch[scripts/functions@2335]
  Error:   >>        called from: CT_PackageRun[scripts/functions@2090]
  Error:   >>        called from: CT_ExtractPatch[scripts/functions@2430]
  Error:   >>        called from: do_cc_extract[scripts/build/cc/gcc.sh@28]
  Error:   >>        called from: main[scripts/crosstool-NG.sh@668]
  Error:   >>
  Error:   >>  For more info on this error, look at the file: 'build.log'
  Error:   >>  There is a list of known issues, some with workarounds, in:
  Error:   >>      https://crosstool-ng.github.io/docs/known-issues/
  Error:   >>
  Error:   >> NOTE: You configuration uses non-default patch sets. Please
  Error:   >> select 'bundled' as the set of patches applied and attempt
  Error:   >> to reproduce this issue. Issues reported with other patch
  Error:   >> set selections (none, local, bundled+local) are going to be
  Error:   >> closed without explanation.
  Error:   >>
  Error:   >>  If you feel this is a bug in crosstool-NG, report it at:
  Error:   >>      https://github.com/crosstool-ng/crosstool-ng/issues/
  Error:   >>
  Error:   >>  Make sure your report includes all the information pertinent to this issue.
  Error:   >>  Read the bug reporting guidelines here:
  Error:   >>      http://crosstool-ng.github.io/support/
  Error:   
  Error:   (elapsed: 6:46.26)
  make[1]: *** [build] Error 1
  /opt/crosstool-ng-latest/bin/ct-ng:261: recipe for target 'build' failed
  make: *** [build.32] Error 2
  /opt/crosstool-ng-latest/bin/ct-ng:265: recipe for target 'build.32' failed
  Error: Process completed with exit code 2.
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -20,8 +20,6 @@ RUN apt-get update -y -q && apt-get upgrade -y -q && apt-get upgrade -y -q && \
     flex \
     git \
     gawk \
-    gcc \
-    g++ \
     binutils-multiarch \
     gperf \
     help2man \
@@ -42,18 +40,20 @@ RUN apt-get update -y -q && apt-get upgrade -y -q && apt-get upgrade -y -q && \
     gettext \
     vim \
     zlib1g-dev \
+    software-properties-common \
     xz-utils && \
     cd /tmp && \
     curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip" && \
     unzip awscliv2.zip && \
     ./aws/install && \
-    rm -rf aws* && \
-    cd /opt && \
-    curl "https://compiler-explorer.s3.amazonaws.com/opt/gcc-11.2.0.tar.xz" -o gcc11.tar.xz && \
-    tar Jxvf gcc11.tar.xz && \
-    rm gcc11.tar.xz
+    rm -rf aws*
 
-ENV PATH="/opt/gcc-11.2.0/bin:${PATH}"
+# Install GCC 11 from the Ubuntu test repository
+RUN add-apt-repository ppa:ubuntu-toolchain-r/test -y && \
+    apt-get update && \
+    apt-get install -y -q gcc-11 g++-11 gnat-11 && \
+    update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-11 60 --slave /usr/bin/g++ g++ /usr/bin/g++-11 && \
+    update-alternatives --config gcc
 
 WORKDIR /opt
 COPY build/patches/cross-tool-ng/cross-tool-ng-1.22.0.patch ./

--- a/build/build.sh
+++ b/build/build.sh
@@ -61,7 +61,7 @@ fi
 
 
 cp "${CONFIG_FILE}" .config
-${CT} oldconfig
+${CT} olddefconfig
 # oldconfig will restore mirror urls, so as a workaround until
 # https://github.com/crosstool-ng/crosstool-ng/issues/1609 gets
 # fixed we have to update the mirror url after calling oldconfig

